File: common/js/google.ts
import type { AgentContext, AgentRequest, AgentResponse } from '@agentuity/sdk';
import { GoogleGenAI } from '@google/genai';

// Get your API key here: https://aistudio.google.com/apikey
if (!process.env.GOOGLE_API_KEY) {
  console.error('Missing the GOOGLE_API_KEY environment variable');

  process.exit(1);
}

const google = new GoogleGenAI({ apiKey: process.env.GOOGLE_API_KEY });

export const welcome = () => {
  return {
    welcome:
      'Welcome to the Google AI TypeScript Agent! I can help you build AI-powered applications using Gemini models.',
    prompts: [
      {
        data: 'How do I implement streaming responses with Gemini models?',
        contentType: 'text/plain',
      },
      {
        data: 'What are the best practices for prompt engineering with Gemini?',
        contentType: 'text/plain',
      },
    ],
  };
};

export default async function Agent(
  req: AgentRequest,
  resp: AgentResponse,
  ctx: AgentContext,
) {
  try {
    const result = await google.models.generateContent({
      model: 'gemini-2.0-flash',
      contents: (await req.data.text()) ?? 'Hello, Gemini',
    });

    return resp.text(result.text ?? 'Something went wrong');
  } catch (error) {
    ctx.logger.error('Error running agent:', error);

    return resp.text('Sorry, there was an error processing your request.');
  }
}

File: common/py/google.py
from google import genai
from agentuity import AgentRequest, AgentResponse, AgentContext
import os

# Get your API key here: https://aistudio.google.com/apikey
api_key = os.getenv("GOOGLE_API_KEY")
if not api_key:
    raise ValueError("GOOGLE_API_KEY environment variable not set.")

client = genai.Client(api_key=api_key)

def welcome():  
    return {  
        "welcome": "Welcome to the Google AI Agent! I can help you interact with Gemini models for natural language tasks.",  
        "prompts": [  
            {  
                "data": "Write a creative story about a journey through time",  
                "contentType": "text/plain"  
            },  
            {  
                "data": "Explain quantum computing to a high school student",  
                "contentType": "text/plain"  
            }  
        ]  
    }

async def run(request: AgentRequest, response: AgentResponse, context: AgentContext):
    
    chat_completion = client.models.generate_content(
        model="gemini-2.0-flash", contents= await request.data.text() or "Why is the sky blue?"
    )
    
    return response.text(chat_completion.text)